compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1729,23 +1748,26 @@ convertDataCon env m con args
                     EVariantCon tcon ctorName $
                     ERecCon (TypeConApp recTCon tyArgs) (zipExact fldNames tmArgs)
 
-    -- Partially applied
-    | otherwise = do
-        fmap (\op -> (EVal op, args)) (qual mkWorkerName (getOccText con))
-
-    where
-
-        qual :: (T.Text -> n) -> T.Text -> ConvertM (Qualified n)
-        qual f t
-            | Just xs <- T.stripPrefix "(," t
-            , T.dropWhile (== ',') xs == ")" = qDA_Types env $ f $ "Tuple" <> T.pack (show $ T.length xs + 1)
-            | IgnoreWorkerPrefix t' <- t = qualify env m $ f t'
+    qual :: (T.Text -> n) -> T.Text -> ConvertM (Qualified n)
+    qual f t
+        | Just xs <- T.stripPrefix "(," t
```

Comment:
this code freaks me out everytime I see it after I’ve forgotten it exists.

Reply:
🙃

Reply:
😮